src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs

```diff
@@ -0,0 +1,220 @@
+        }
+
+        public async Task<QueryResult<IList<KnownEndpointsView>>> QueryKnownEndpoints()
+        {
+            var knownEndpoints = new List<KnownEndpointsView>();
+
+            await connectionManager.Perform(async connection =>
+            {
+                var rows = await connection.QueryAsync("SELECT * FROM [MessagesView]").ConfigureAwait(false);
+
+                foreach (var row in rows)
+                {
+                    knownEndpoints.Add(new KnownEndpointsView
+                    {
+                        Id = row.Id,
+                        HostDisplayName = row.HostDisplayName
+                        //EndpointDetails = row.EndpointDetails
+                    });
+                }
+            }).ConfigureAwait(false);
+
+            return new QueryResult<IList<KnownEndpointsView>>(knownEndpoints, new QueryStatsInfo(string.Empty, knownEndpoints.Count));
+        }
+
+        public Task<QueryResult<IList<MessagesView>>> QueryMessages(string searchParam, PagingInfo pagingInfo, SortInfo sortInfo) => throw new NotImplementedException();
+        public async Task<QueryResult<IList<MessagesView>>> QueryMessagesByConversationId(string conversationId, PagingInfo pagingInfo, SortInfo sortInfo)
+        {
+            var messagesViews = new List<MessagesView>();
+
+            await connectionManager.Perform(async connection =>
+            {
+                var rows = await connection.QueryAsync(
+                    @"SELECT * FROM [dbo].[MessagesView] WHERE [ConversationId] = @Id",
+                    new
+                    {
+                        Id = conversationId
+                    }).ConfigureAwait(false);
+
+                foreach (var row in rows)
+                {
+                    messagesViews.Add(new MessagesView
+                    {
+                        BodySize = row.BodySize,
+                        BodyUrl = row.BodyUrl,
+                        ConversationId = row.ConversationId,
+                        CriticalTime = row.CriticalTime,
+                        DeliveryTime = row.DeliveryTime,
+                        Id = row.Id,
+                        InstanceId = row.InstanceId,
+                        IsSystemMessage = row.IsSystemMessage,
+                        MessageId = row.MessageId,
+                        MessageIntent = row.MessageIntent,
+                        MessageType = row.MessageType,
+                        ProcessedAt = row.ProcessedAt,
+                        ProcessingTime = row.ProcessingTime,
+                        Status = row.Status,
+                        TimeSent = row.TimeSent
+                        //Headers = row.Headers,
+                        //InvokedSagas = row.InvokedSagas,
+                        //OriginatesFromSaga = row.OriginatesFromSaga,
+                        //ReceivingEndpoint = row.ReceivingEndpoint,
+                        //SendingEndpoint = row.SendingEndpoint
```

Comment:
Remove?

Reply:
It's what's left to do so I've added TODO comments above those to make it clear